### PR TITLE
Adds and improves logging for various shit.

### LINKED ISSUE #68452: No logs when a frozen obj is shattered
## Reproduction:

1. Freeze a head
2. throw it
3. it shatters with no admin logs

### LINKED ISSUE #71798: Records consoles (Security, Med) have no logging for name changes
## Testmerges:

n/a

## Reproduction:
There is no logging (in-round or administrative) for anyone changing the records in records consoles (notably the security one since the medical records only have flavour value)
You can change everyone's name to dickbutt on their record and admins arent gonna know for sure who done it

### LINKED ISSUE #78008: Flamethrower logging in attack logs despite not being ignited
Reporting client version: 514.1589

## Round ID:
[213658](https://scrubby.melonmesa.com/round/213658)

## Reproduction:
So using a flamethrower logs as "flamethrowered [x] with the flamethrower". This event gets logged the same no matter if the flamethrower is lit or not, leading to potential confusions while proofreading logs. Having a check for whether the flamethrower is lit would be appreciated.

### LINKED ISSUE #81098: Poor logging of a combat event
Reporting client version: 514.1589

## Round ID:
[222727](https://scrubby.melonmesa.com/round/222727)

## Testmerges:
- [Temp keybinds patch for 515.1628](https://www.github.com/tgstation/tgstation/pull/81089)

## Reproduction:

```
[2024-01-24 19:52:36.155] ATTACK: Thekillerchef/(Chadwick Nelson) (mob_4307) has been shot by  with the bolt (Central Primary Hallway (92,145,2))
```

Not really sure what this means, i don't find any adjacent cause of "a bolt" in the combat logs, the firer seems missing, there's not even a damage mark on this line to tell if it did anything?

### LINKED ISSUE #81130: RCD logging doesn't seem to exist?
## Round ID:

None specific - a current round is 222886

## Reproduction:

Unless me and several other admins are missing something, RCD actions dont seem to be logged.  Since the proliferation of these in the engineering start kit they're periodically used as deconstruction tiding/griefing (esp on multi-z), or grief construction of obstructions, and while we'd like to act on some of this there's a lack of audit trail for such things.

Please ensure that all construction and deconstructions are logged somewhere, ideally also on the player panel's game or attack section (whatever is considered normal).

Thanks

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
Adds logging for RCD construction and deconstruction. Hallucinated projectiles no longer causes logs. Flamethrowers log gas mixture information, the flamethrower, the gas tank, tank distribution pressure and whether it was lit. Adds a lot more logging to records consoles. Frozen objects now log when they get shattered. 
## Why It's Good For The Game
Closes #68452 
Closes #71798 
Closes #78008 
Closes #81098 
Closes #81130 
## Changelog
:cl: Pickle-Coding and Rhials
admin: RCD construction and deconstruction are logged.
admin: Hallucinated projectiles no longer log.
admin: Gives more detail to flamethrower logging.
admin: More actions are logged for records consoles usage.
admin: Frozen object shattering is logged.
/:cl:

### CODE REVIEW

File: code/game/objects/items/rcd/RCD.dm [hunk 1]
```diff
@@ -221,6 +221,7 @@
 	current_active_effects += 1
 	_rcd_create_effect(target, user, delay, rcd_results)
 	current_active_effects -= 1
+	log_tool("used RCD with design mode: \"[rcd_results["[RCD_DESIGN_MODE]"]]\" and design path: \"[rcd_results["[RCD_DESIGN_PATH]"]]\" with delay [delay]", user)
```
Comment (SyncIt21) [inline code/game/objects/items/rcd/RCD.dm:224]: Makes sense to log only if the effect was successful.

Luckily _rcd_create_effect() proc returns TRUE if it was so you can write an if condition there and log only if that proc returns TRUE.

File: code/__HELPERS/logging/atmos.dm [hunk 2]
```diff
@@ -1,8 +1,5 @@
 /// Logs the contents of the gasmix to the game log, prefixed by text
```
Comment (ZephyrTFA) [inline code/__HELPERS/logging/atmos.dm:1]: make this use the data list

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

File: code/__HELPERS/logging/atmos.dm (hunk 2 above)
Comment (Pickle-Coding) [reply code/__HELPERS/logging/atmos.dm:1]: What is the data list used for?

Comment (SyncIt21) [reply code/__HELPERS/logging/atmos.dm:1]: The data list is your user defined map of key(string) to value(can be anything  but is converted to string anyway) which is written in json format to your log file.

The name of your file where this data is stored is <our usual log file>-json.json.  obviously json format can be parsed later for statictics and what not

In your case vars like temperature and moles can be put in list format as 2 seperate keys and passed to log game.

Message should only be used to explain what vars are being logged and not store the actual data itself

Comment (Pickle-Coding) [reply code/__HELPERS/logging/atmos.dm:1]: Done.

**Comment by Pickle-Coding** [comment]
Drafting because there are a few issues with my PR that I'll fix later.

**Review by ZephyrTFA — APPROVED** [review]

**Comment by ZephyrTFA** [comment]
go away stalebot

**Pull request merged**